pkg/integration/utils.go

```diff
@@ -344,6 +344,21 @@ func (c *ChannelBuffer) ReadTimeout(p []byte, n time.Duration) (int, error) {
 	}
 }
 
+// FlushTimeout flush content of channel within the specified duration as timeout
+func (c *ChannelBuffer) FlushTimeout(n time.Duration) error {
+	var stop bool
+	go func() {
+		<-time.After(n)
+		stop = true
+	}()
+	for range c.C {
+		if stop {
```

Comment:
Reading and writing to the same variable without synchronization will be racey and could cause a panic.
Also not sure what this is flush is doing, I don't see where we are flushing.